core/src/main/java/org/apache/calcite/avatica/MetaImpl.java

```diff
@@ -1527,12 +1527,12 @@ public Object next() {
 
   /** Iterable that yields an iterator over rows coming from a sequence of
    * {@link Meta.Frame}s. */
-  private class FetchIterable implements Iterable<Object> {
```

Comment:
Changing the visibility on these implementation details suggests to me they should probably be top-level classes.